internal/cli/daemon/daemon.go

```diff
@@ -58,14 +58,25 @@ func NewCommand(srv rpc.ArduinoCoreServiceServer, settings *rpc.Configuration) *
 				})
 				if err != nil {
 					// Should never happen...
-					panic("Failed to set default value for directories.builtin.libraries: " + err.Error())
+					feedback.Fatal(i18n.Tr("Cannot set the configuration key directories.builtin.libraries: %v", err), feedback.ErrGeneric)
 				}
 			}
 
 			// Validate the maxGRPCRecvMsgSize flag
 			if maxGRPCRecvMsgSize < 1024 {
 				feedback.Fatal(i18n.Tr("%s must be >= 1024", "--max-grpc-recv-message-size"), feedback.ErrBadArgument)
 			}
+
+			// The user agent should include "daemon" for analytics purposes
+			_, err := srv.SettingsSetValue(cmd.Context(), &rpc.SettingsSetValueRequest{
```

Comment:
> The user agent should include "daemon" for analytics purposes

Actually no, it should infer the user agent from the gRPC client:

https://github.com/arduino/arduino-cli/blob/6d9c930c79b6e8609915da0b774680588c5cc363/commands/instances.go#L67-L70

I think the Arduino IDE is passing something like `arduino-ide/x.y.z` into this string.
We should understand why this string is not propagated.